src/classes/RD2_DataMigration_TEST.cls

```diff
@@ -131,6 +131,54 @@ private class RD2_DataMigration_TEST {
             'This RD should be in Enhanced RD format');
     }
 
+     /**
+     * @description Verifies that a schedule record is created for Enhanced Recurring Donation record.
+     */
+    @IsTest
+    private static void shouldCreateScheduleForActiveEnhancedRDs() {
+        List<RecurringDonationSchedule__c> rdSchedules  = new List<RecurringDonationSchedule__c>();
+        Id contactId = UTIL_UnitTestData_TEST.mockId(Contact.SObjectType);
+        npe03__Recurring_Donation__c rd = getNewRecurringDonationEnhancedFormat(contactId);
+
+        RD2_DataMigrationMapper mapper = new RD2_DataMigrationMapper(rd);
+        mapper.convertToEnhancedRD();
+
+        rdSchedules = mapper.getScheduleRecords();
+
+        System.assertEquals(1, rdSchedules.size(), 'Number of RDs should be 1');
+        System.assertEquals(100, rdSchedules[0].InstallmentAmount__c, 'Installment Amount should be 100');
+        System.assertEquals('1', rdSchedules[0].DayOfMonth__c,  'Day of Month should be 1');
+    }
+
+    /**
+     * @description Verifies that a schedule record is not created for Closed Enhanced Recurring Donation record.
+     */
+    @IsTest
+    private static void shouldNotCreateScheduleForClosedEnhancedRDs() {
+
+        List<RecurringDonationSchedule__c> rdSchedules  = new List<RecurringDonationSchedule__c>();
+
+        npe03__Recurring_Donation__c rds =
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withMockId()
+                .withContact(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
+                .withAmount(400)
+                .withPaymentMethod('Check')
+                .withInstallmentPeriodWeekly()
+                .withInstallmentFrequency(1)
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
```

Comment:
Why these particular dates and numbers? Can we reuse the RD construction as done in the previous test but just set Open Ended Status to Closed before RD is being converted?

Reply:
I am using the new method .withRecurringDonationStatusClosed() in that class to mark the status as closed, Do you want to change it?